test/src/main/java/org/apache/accumulo/test/functional/ScannerContextIT.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.util.CachedConfiguration;
+import org.apache.accumulo.fate.util.UtilWaitThread;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.apache.accumulo.minicluster.impl.MiniAccumuloClusterImpl;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ScannerContextIT extends AccumuloClusterHarness {
+
+  private static final String CONTEXT = ScannerContextIT.class.getSimpleName();
+  private static final String CONTEXT_PROPERTY = Property.VFS_CONTEXT_CLASSPATH_PROPERTY + CONTEXT;
+  private static final String CONTEXT_DIR = "file:///tmp";
+  private static final String CONTEXT_CLASSPATH = CONTEXT_DIR + "/Test.jar";
+  private static int ITERATIONS = 10;
+
+  @Override
+  protected int defaultTimeoutSeconds() {
+    return 2 * 60;
+  }
+
+  @Before
+  public void checkCluster() {
+    Assume.assumeThat(getClusterType(), CoreMatchers.is(ClusterType.MINI));
+    MiniAccumuloClusterImpl mac = (MiniAccumuloClusterImpl) getCluster();
+  }
+
+  @Test
+  public void test() throws Exception {
+    // Copy the TestIterators jar to tmp
+    FileSystem fs = FileSystem.get(CachedConfiguration.getInstance());
+    Path srcPath = new Path(System.getProperty("user.dir") + "/target/TestIterators-tests.jar");
+    Path dstPath = new Path(CONTEXT_DIR + "/Test.jar");
+    fs.copyFromLocalFile(srcPath, dstPath);
+    // Sleep to ensure jar change gets picked up
+    UtilWaitThread.sleep(5000);
+
+    try {
+      Connector c = getConnector();
+      // Set the classloader context property on the table to point to the TestIterators jar file.
+      c.instanceOperations().setProperty(CONTEXT_PROPERTY, CONTEXT_CLASSPATH);
+
+      // Insert rows with the word "Test" in the value.
+      String tableName = getUniqueNames(1)[0];
+      c.tableOperations().create(tableName);
+      BatchWriter bw = c.createBatchWriter(tableName, new BatchWriterConfig());
+      for (int i = 0; i < ITERATIONS; i++) {
+        Mutation m = new Mutation("row" + i);
+        m.put("cf", "col1", "Test");
+        bw.addMutation(m);
+      }
+      bw.close();
+      // Ensure that we can get the data back
+      scanCheck(c, tableName, null, null, "Test");
+      batchCheck(c, tableName, null, null, "Test");
+
+      // This iterator is in the TestIterators jar file
+      IteratorSetting cfg = new IteratorSetting(1, "reverse", "org.apache.accumulo.test.functional.ValueReversingIterator");
```

Comment:
Should follow best-practices and use a reasonable priority (e.g. > 20). Would also be better to reference the `ValueReversingIterator.class` for IDEs to automatically handle potential renames in the future.